heed/src/cookbook.rs

```diff
@@ -0,0 +1,450 @@
+//! A cookbook of examples on how to use heed. Here is the list of the different topics you can learn about:
+//!
+//! - [Listing and Opening the Named Databases](#listing-and-opening-the-named-databases)
+//! - [Create Custom and Prefix Codecs](#create-custom-and-prefix-codecs)
+//! - [Change the Environment Size Dynamically](#change-the-environment-size-dynamically)
+//! - [Decode Values on Demand](#decode-values-on-demand)
+//! - [Advanced Multithreaded Access of Entries](#advanced-multithreaded-access-of-entries)
+//!
+//! # Listing and Opening the Named Databases
+//!
+//! Sometimes it is useful to list the databases available in an environment.
+//! LMDB stores them already in the unnamed database, a database that is always
+//! there in which you can write.
+//!
+//! However, once you create new databases, after defining the [`EnvOpenOptions::max_dbs`]
+//! parameter, the names of those databases are automatically stored in the unnamed one.
+//!
+//! ```
+//! use std::error::Error;
+//! use std::fs;
+//! use std::path::Path;
+//!
+//! use heed::types::*;
+//! use heed::{Database, EnvOpenOptions};
+//!
+//! fn main() -> Result<(), Box<dyn Error>> {
+//!     let env_path = Path::new("target").join("heed.mdb");
+//!
+//!     fs::create_dir_all(&env_path)?;
+//!
+//!     let env = unsafe {
+//!         EnvOpenOptions::new()
+//!             .map_size(10 * 1024 * 1024) // 10MB
+//!             .max_dbs(3) // Number of opened databases
+//!             .open(env_path)?
+//!     };
+//!
+//!     let rtxn = env.read_txn()?;
+//!     let unnamed: Database<Str, DecodeIgnore> =
+//!         env.open_database(&rtxn, None)?.expect("the unnamed database always exists");
+//!
+//!     // The unnamed (or main) database contains the other
+//!     // database names associated to empty values.
+//!     for result in unnamed.iter(&rtxn)? {
+//!         let (name, ()) = result?;
+//!
+//!         if let Ok(Some(_db)) = env.open_database::<Str, Bytes>(&rtxn, Some(name)) {
+//!             // We succeeded into opening a new database that
+//!             // contains strings associated to raw bytes.
+//!         }
+//!     }
+//!
+//!     // When opening databases in a read-only transaction
+//!     // you must commit your read transaction to make your
+//!     // freshly opened databases globally available.
+//!     rtxn.commit()?;
+//!
+//!     // If you abort (or drop) your read-only transaction
+//!     // the database handle will be invalid outside
+//!     // the transaction scope.
+//!
+//!     Ok(())
+//! }
+//! ```
+//!
+//! # Create Custom and Prefix Codecs
+//!
+//! With heed you can store any kind of data and serialize it the way you want.
+//! To do so you'll need to create a codec by using the [`BytesEncode`] and [`BytesDecode`] traits.
+//!
+//! Now imagine that your data is lexicographically well ordered. You can now leverage
+//! the use of prefix codecs. Those are classic codecs but are only used to encode key prefixes.
```

Comment:
I find the original sentence a bit hard to understand.
1. Let me know if I’m wrong, but your data is _always_ lexicographically; thus, I would prefer to avoid any possible confusion with the « now imagine »
2. Maybe a tiny example could help understand what a prefix is?

```suggestion
//! Since data are always lexicographically ordered in LMDB. You can create
//! prefix codecs. Those are classic codecs but are only used to encode key prefixes.
//! For example, the prefix `the` could retrieve the keys `the`, `their`, `them`, and `there` in this order.
```